frappe_graphql/utils/generate_sdl/doctype.py

```diff
@@ -160,6 +180,12 @@ def get_graphql_type(docfield):
         graphql_type = "BaseDocType"
     elif docfield.fieldtype in table_fields:
         graphql_type = f"[{docfield.options.replace(' ', '')}!]!"
+    elif docfield.fieldtype == "Select":
+        graphql_type = get_select_docfield_enum_name(meta.name, docfield)
+
+        # Mark NonNull if there is no empty option
+        if all([len(x or "") > 0 for x in (docfield.options or "").split("\n")]):
+            graphql_type += "!"
```

Comment:
hmmm i think only check meta if the select `docfield` **is_mandatory** and append the `!`

Reply:
resolved 👍